packages/compass-import-export/src/components/import-preview.tsx

```diff
@@ -47,49 +59,101 @@ const cellUncheckedStyles = css({
-  setFieldType,
-  loaded,
+const rowIndexStyles = css({
+  minWidth: 0,
+  color: palette.gray.base,
+});
+
+const fieldTypeContainerStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[1],
+});
+
+const infoIconCSS = css({
+  height: spacing[3],
+  color: palette.gray.base,
+});
+
+type Field = {
+  path: string;
+  type: CSVParsableFieldType | 'placeholder';
+  checked: boolean;
+  result?: CSVField;
+};
+
+function MixedWarning({
+  result,
+  selectedType,
 }: {
-  fields: {
-    path: string;
-    type: string;
-    checked: boolean;
-  }[];
-  values: string[][];
-  onFieldCheckedChanged: (fieldPath: string, checked: boolean) => void;
-  setFieldType: (fieldPath: string, fieldType: string) => void;
-  loaded: boolean;
+  result: CSVField;
+  selectedType: CSVParsableFieldType;
 }) {
-  if (!loaded) {
-    debug('Preview unavailable: not loaded yet');
+  if (!['mixed', 'number'].includes(result.detected)) {
     return null;
   }
 
-  if (!Array.isArray(fields) || !Array.isArray(values)) {
-    debug('Preview unavailable: Fields or values is not an array', {
-      fields,
-      values,
-    });
+  // once the user manually changed the type, make the warning go away
+  if (result.detected !== selectedType) {
     return null;
   }
 
   return (
-    <>
-      <Body weight="medium">Specify Fields and Types</Body>
-      <Table
-        data={values}
-        columns={fields.map((field) => (
-          <TableHeader
-            key={`col-${field.path}`}
-            label={
-              <div
-                className={columnHeaderStyles}
-                data-testid={`preview-field-header-${field.path}`}
-              >
+    <Tooltip
+      align="top"
+      justify="middle"
+      delay={500}
+      trigger={({ children, ...props }) => (
+        <div {...props}>
+          {children}
+          <div className={infoIconCSS}>
+            <Icon glyph="InfoWithCircle"></Icon>
+          </div>
+        </div>
+      )}
+    >
+      <>
+        <p>
+          This field has{' '}
+          {selectedType === 'number'
+            ? 'mixed numeric types'
+            : 'mixed data types'}
+          :
+        </p>
+        <ul>
+          {Object.entries(result.types).map(([type, info]) => {
+            return (
+              <li key={type}>
+                {type} * {info.count}
+              </li>
+            );
+          })}
+        </ul>
+        <p>To standardise your data, select a different type.</p>
+      </>
+    </Tooltip>
+  );
+}
+
+function FieldHeader(
+  field: Field,
+  analyzed: boolean,
+  onFieldCheckedChanged: (fieldPath: string, checked: boolean) => void,
+  setFieldType: (fieldPath: string, fieldType: string) => void
+) {
+  return (
+    <TableHeader
+      key={`col-${field.path}`}
+      label={
+        <div
+          className={columnHeaderStyles}
+          data-testid={`preview-field-header-${field.path}`}
+        >
+          {field.type === 'placeholder' ? (
```

Comment:
All this mention of 'placeholder' relates to when there are arrays in the CSV. We won't be exporting stuff like this until export is changed, but import now already supports CSV files like that.

<img width="991" alt="Screenshot 2023-03-17 at 14 41 47" src="https://user-images.githubusercontent.com/69737/225937058-d0b75701-5a18-4b31-89a9-999325652ffc.png">

Here's the file that produced that:

```
a,foo[0],foo[1],foo[2],bar[0][0],bar[0][1],z,notes
a,1,2,3,,,z,foo is an array of simple values
a,1,,,,,z,foo is not always of the same length
```

See the columns in the preview still match one to one with the cells in the CSV file, but for arrays the type is for the values inside the array and the whole array gets one type.

Reply:
Probably not for this iteration, but I could see folks sometimes having `placeholder` items in arrays like this. We might want a way to collapse/expand those fields so the table is still scrollable. This also goes back to your comments on the UX of having this table format for negotiating the import.

Reply:
I'm not sure how much we should worry about it because you can only get into that situation when re-importing CSV we exported. Which we should absolutely support, but we'd also be warning people NOT to use CSV for that sort of thing. If you're going to export it only to re-import back into mongodb, then extended JSON is obviously a much better format.

So in reality CSV should be for when exporting for use with a third party system or when importing from some third party system.

Reply:
Sounds good 👍 
I wonder if we should start recommending people the json export type when we do the export work.

Reply:
We should. It is in the design :)